iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/TsFileResource.java

```diff
@@ -38,71 +47,90 @@
 public class TsFileResource {
 
   private OverflowChangeType overflowChangeType;
+
+  //the file index of `settled` folder in the Directories.
   private int baseDirIndex;
-  private String relativePath;
+  private File file;
   private Map<String, Long> startTimeMap;
   private Map<String, Long> endTimeMap;
   private Set<String> mergeChanged = new HashSet<>();
 
   private transient ModificationFile modFile;
 
-  public TsFileResource(Map<String, Long> startTimeMap, Map<String, Long> endTimeMap,
-      OverflowChangeType type, int baseDirIndex, String relativePath) {
-
-    this.overflowChangeType = type;
-    this.baseDirIndex = baseDirIndex;
-    this.relativePath = relativePath;
-
-    this.startTimeMap = startTimeMap;
-    this.endTimeMap = endTimeMap;
-    this.modFile = new ModificationFile(
-        Directories.getInstance().getTsFileFolder(baseDirIndex) + File.separator
-            + relativePath + ModificationFile.FILE_SUFFIX);
+  /**
+   * @param autoRead whether read the file to initialize startTimeMap and endTimeMap
+   */
+  public TsFileResource(File file, boolean autoRead) throws IOException {
```

Comment:
BaseDirIndex is not init is this constructor? Won't it be used later?

Reply:
I init it in `public TsFileResource(Map<String, Long> startTimeMap, Map<String, Long> endTimeMap,
      OverflowChangeType type, File file)` which is called by this constructor.